cmd/kubeadm/app/cmd/init.go

```diff
@@ -155,11 +160,22 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 		},
 	}
 
-	// adds command flags
-	AddInitConfigFlags(cmd.PersistentFlags(), options.externalcfg, &options.featureGatesString)
-	AddInitOtherFlags(cmd.PersistentFlags(), &options.cfgPath, &options.skipTokenPrint, &options.dryRun, &options.ignorePreflightErrors)
-	options.bto.AddTokenFlag(cmd.PersistentFlags())
-	options.bto.AddTTLFlag(cmd.PersistentFlags())
+	// adds flags to the init command
+	// init command local flags could be eventually inherited by the sub-commands automatically generated for phases
+	AddInitConfigFlags(cmd.Flags(), options.externalcfg, &options.featureGatesString)
+	AddInitOtherFlags(cmd.Flags(), &options.cfgPath, &options.skipTokenPrint, &options.dryRun, &options.ignorePreflightErrors)
+	options.bto.AddTokenFlag(cmd.Flags())
+	options.bto.AddTTLFlag(cmd.Flags())
+
+	// defines additional flag that are not used by the init command but that could be eventually used
+	// by the sub-commands automatically generated for phases
+	initRunner.SetPhaseSubcommandsAdditionalFlags(func(flags *flag.FlagSet) {
+		flags.Var(utilflag.NewMapStringString(&options.externalcfg.APIServerExtraArgs), "apiserver-extra-args", "A set of extra flags to pass to the API Server or override default ones in form of <flagname>=<value>")
```

Comment:
could we move all these flags into functions? some already exist at https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/options/generic.go and it'd be best to have the prose in one place